azure/services/agentpools/agentpools.go

```diff
@@ -149,6 +153,17 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	return nil
 }
 
+// get custom headers to pass to agent pool client from agent pool annotations.
+func (s *Service) getCustomHeaders() map[string]string {
+	var result = map[string]string{}
+	for key, value := range s.scope.AgentPoolAnnotations() {
+		if strings.HasPrefix(key, azure.CustomHeaderPrefix) {
+			result[strings.TrimPrefix(key, azure.CustomHeaderPrefix)] = value
```

Comment:
Just a thought: 
What if the value is empty? We can have probably move this function to some better place that takes input as `annotations` and `prefix` and returns the filtered values.  This could help reuse. 

And here we can probably just sanitise the values if required.

Reply:
Refactored for reuse. What issue do you see with an empty value? We'll just pass a header with empty value if that is what someone intends.

Reply:
Sure. My only point was -- does the azure api tolerates the empty value. I do not know. Just thought to bring up, in case. If yes, I do not think we need to do anything here.

Reply:
Sure.